scipy/integrate/tests/test_quadrature.py

```diff
@@ -325,6 +327,61 @@ def test_masked(self):
         assert_allclose(trapezoid(y, xm), r)
 
 
+@array_api_compatible
+class TestSimpson:
+    def test_simple(self, xp):
+        x = xp.arange(-10, 10, .1)
+        r = simpson(xp.exp(-.5 * x ** 2) / xp.sqrt(2 * xp.asarray(xp.pi)), dx=0.1)
+        # check integral of normal equals 1
+        xp_assert_close(r, xp.asarray(1.0))
+
+    def test_ndim(self, xp):
+        x = xp.linspace(0, 1, 3)
+        y = xp.linspace(0, 2, 8)
+        z = xp.linspace(0, 3, 13)
```

Comment:
This seems to fix the `torch` issue locally (I guess you're going to replace the tests anyway) by allowing the expected value to overcome the default `dtype` used by `torch`.

```diff
--- a/scipy/integrate/tests/test_quadrature.py
+++ b/scipy/integrate/tests/test_quadrature.py
@@ -336,9 +336,9 @@ class TestSimpson:
         xp_assert_close(r, xp.asarray(1.0))
 
     def test_ndim(self, xp):
-        x = xp.linspace(0, 1, 3)
-        y = xp.linspace(0, 2, 8)
-        z = xp.linspace(0, 3, 13)
+        x = xp.linspace(0, 1, 3, dtype=xp.float64)
+        y = xp.linspace(0, 2, 8, dtype=xp.float64)
+        z = xp.linspace(0, 3, 13, dtype=xp.float64)
```